Lib/subprocess.py

```diff
@@ -1480,7 +1480,19 @@ def _execute_child(self, args, executable, preexec_fn, close_fds,
             if shell:
                 startupinfo.dwFlags |= _winapi.STARTF_USESHOWWINDOW
                 startupinfo.wShowWindow = _winapi.SW_HIDE
-                comspec = os.environ.get("COMSPEC", "cmd.exe")
+                # gh-101283: with no full path, Windows looks into a
+                # current directory first so no plain "cmd.exe".
+                system_drive = os.environ.get('SystemDrive') or 'C:'
+                system_root = os.environ.get('SystemRoot') or os.path.join(
+                    system_drive, 'Windows')
+                comspec = os.environ.get('ComSpec') or os.path.join(
+                    system_root, 'System32', 'cmd.exe')
+                if not os.path.isfile(comspec):
+                    # Windows is installed into a non-standard location
+                    # or the system environment variables are broken.
+                    # It's highly unlikely and we cannot help here.
+                    comspec = "cmd.exe"
```

Comment:
Starting a process is already expensive on Windows. We should avoid the cost of `os.stat()` in the common case. We always use the `ComSpec` environment variable if it's defined and not an empty string. The fallback to "%SystemRoot%\\System32\\cmd.exe" or "%SystemDrive%\\Windows\\System32\\cmd.exe" should be checked before falling back on "cmd.exe". (Alternatively, the `_winapi.CreateProcess()` call could be retried if it fails with `FileNotFoundError` with `shell=True`, but that's I think needlessly complicated for this fallback case. I just don't want to slow down the common, expected path.)

Also, if `comspec` is a fully qualified path and `executable` isn't set, we can set `executable` to let `CreateProcessW()` skip its internal `SearchPathW()` call.

```suggestion
                comspec = os.environ.get('ComSpec')
                if not comspec:
                    system_drive = os.environ.get('SystemDrive') or 'C:'
                    system_root = os.environ.get('SystemRoot') or os.path.join(
                        system_drive, 'Windows')
                    comspec = os.path.join(system_root, 'System32', 'cmd.exe')
                    if not os.path.isfile(comspec):
                        # Windows is installed into a non-standard location
                        # or the system environment variables are broken.
                        # It's highly unlikely and we cannot help here.
                        comspec = 'cmd.exe'
                if not executable and os.path.isabs(comspec):
                    executable = comspec
```